zkevm-circuits/src/evm_circuit/step.rs

```diff
@@ -0,0 +1,553 @@
+    pub(crate) call_id: Cell<F>,
+    pub(crate) is_root: Cell<F>,
+    pub(crate) is_create: Cell<F>,
+    pub(crate) opcode_source: Cell<F>,
+    pub(crate) program_counter: Cell<F>,
+    pub(crate) stack_pointer: Cell<F>,
+    pub(crate) gas_left: Cell<F>,
+    pub(crate) memory_size: Cell<F>,
+    pub(crate) state_write_counter: Cell<F>,
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct StepRow<F> {
+    pub(crate) qs_byte_lookup: Cell<F>,
+    pub(crate) cells: [Cell<F>; STEP_WIDTH],
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct Step<F> {
+    pub(crate) state: StepState<F>,
+    pub(crate) rows: Vec<StepRow<F>>,
+}
+
+impl<F: FieldExt> Step<F> {
+    pub(crate) fn new(
+        meta: &mut ConstraintSystem<F>,
+        qs_byte_lookup: Column<Advice>,
+        advices: [Column<Advice>; STEP_WIDTH],
+        is_next_step: bool,
+    ) -> Self {
+        let state = {
+            let mut cells =
+                VecDeque::with_capacity(ExecutionResult::amount() + 10);
+            meta.create_gate("Query state for step", |meta| {
+                for idx in 0..cells.capacity() {
+                    let column_idx = idx % STEP_WIDTH;
+                    let rotation = idx / STEP_WIDTH
+                        + if is_next_step { STEP_HEIGHT } else { 0 };
+                    cells.push_back(Cell::new(
+                        meta,
+                        advices[column_idx],
+                        rotation,
+                    ));
+                }
+
+                vec![0.expr()]
+            });
+
+            StepState {
+                execution_result: cells
+                    .drain(..ExecutionResult::amount())
+                    .collect(),
+                rw_counter: cells.pop_front().unwrap(),
+                call_id: cells.pop_front().unwrap(),
+                is_root: cells.pop_front().unwrap(),
+                is_create: cells.pop_front().unwrap(),
+                opcode_source: cells.pop_front().unwrap(),
+                program_counter: cells.pop_front().unwrap(),
+                stack_pointer: cells.pop_front().unwrap(),
+                gas_left: cells.pop_front().unwrap(),
+                memory_size: cells.pop_front().unwrap(),
+                state_write_counter: cells.pop_front().unwrap(),
+            }
+        };
+
+        let mut rows = Vec::with_capacity(STEP_HEIGHT - 4);
+        meta.create_gate("Query rows for step", |meta| {
+            for rotation in 4..STEP_HEIGHT {
+                let rotation =
+                    rotation + if is_next_step { STEP_HEIGHT } else { 0 };
+                rows.push(StepRow {
+                    qs_byte_lookup: Cell::new(meta, qs_byte_lookup, rotation),
+                    cells: advices
+                        .map(|column| Cell::new(meta, column, rotation)),
+                });
+            }
+
+            vec![0.expr()]
+        });
+
+        Self { state, rows }
+    }
+
+    pub(crate) fn q_execution_result(
```

Comment:
```suggestion
    pub(crate) fn execution_result_selector(
```

Reply:
Fixed in [`57b71f2`](https://github.com/appliedzkp/zkevm-circuits/pull/196/commits/57b71f29bafa5234bbc8671cc2f7d079e93e7769)